Main.hs

```diff
@@ -0,0 +1,190 @@
+module Main where
+
+import Control.Monad (unless)
+import Text.Printf (printf)
+
+short :: [a] -> Bool
+short (_:_:_:_) = False
+short a = True
+
+lovely :: [Int] -> Bool
+lovely xs = short xs || xs !! 2 == 14
+
+rightTriangles :: [(Int, Int, Int)]
+rightTriangles = [(a, b, c) | a <-[1..], b <-[a..], let c=floor (sqrt (fromIntegral (a^2 + b^2))), a^2 + b^2 == c^2] 
+
+fizzBuzz :: [String]
+fizzBuzz = [fizzBuzz n | n <- [1..]]
+  where 
+    fizzBuzz n
+      | n `mod` 15 == 0 = "FizzBuzz"
+      | n `mod` 3 == 0 = "Fizz"
+      | n `mod` 5 == 0 = "Buzz"
+      | otherwise = show n
+
+ageOn :: String -> Float -> Float
+ageOn planet ageInSeconds = case planet of
+  "Mercury"-> ageInSeconds / (0.2408467 * 31557600)
+  "Venus"-> ageInSeconds / (0.61519726 * 31557600)
+  "Earth"-> ageInSeconds / 31557600
+  "Mars"-> ageInSeconds / (1.8808158 * 31557600)
+  "Jupiter"-> ageInSeconds / (11.862615 * 31557600)
+  "Saturn"-> ageInSeconds / (29.447498 * 31557600)
+  "Uranus"-> ageInSeconds / (84.016846 * 31557600)
+  "Neptune"-> ageInSeconds / (164.79132 * 31557600)
```

Comment:
There is a lot of code duplication here. Avoid it by introducing a function to do this computation.